app/src/main/java/swati4star/createpdf/fragment/PdfToImageFragment.java

```diff
@@ -177,13 +183,57 @@ public void onActivityResult(int requestCode, int resultCode, Intent data) throw
     }
 
     /**
-     * invokes generation of images for pdf pages in the background
+     * invokes generation of images for pdf pages in the background by checking
+     * for encryption first.
      */
     @OnClick(R.id.createImages)
     public void parse() {
-        if (mOperation.equals(PDF_TO_IMAGES))
-            new PdfToImages(mPath, mUri, this).execute();
-        else
+        if (mPDFUtils.isPDFEncrypted(mPath)) {
+            mInputPassword = new String[1];
+            new MaterialDialog.Builder(mActivity)
+                    .title(R.string.enter_password)
+                    .content(R.string.decrypt_message)
```

Comment:
Change content text to `Selected pdf is password protected. Enter the password to continue...`